src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -182,6 +179,28 @@ public PersistentDataStore() {
 
     return  aboutmemessages;
   }
+
+  public List<Statistic> loadStatistics(Statistic.Type type) throws PersistentDataStoreException {
+    List<Statistic> statistics = new ArrayList<>();
+
+    // Retrieve all statistics of that type from the datastore.
+    Query query = new Query(type.getName()).addSort("creation_time", SortDirection.ASCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      try {
+        UUID id = UUID.fromString((String) entity.getProperty("uuid"));
+        long value = (long) entity.getProperty("value");
+        Instant creationTime = Instant.parse((String) entity.getProperty("creation_time"));
+        Statistic statistic = new Statistic(id, creationTime, type, value);
+        statistics.add(statistic);
+      } catch (Exception e) {
+        throw new PersistentDataStoreException(e);
+      }
+    }
+
+    return statistics;
+  }
```

Comment:
You should make a test for this function.

Reply:
I've added a test for this function and for other stuff I implemented! I didn't create a test for the statistics page because I don't think there are any tests for JSPs (in the code they provide us), not sure how I would implement that.